_posts/2020-08-05-using-open-policy-agent-with-strimzi-and-apache-kafka.md

```diff
@@ -0,0 +1,456 @@
+OPA will evaluate the policy and respond to the authorizer request with a decision.
+And the authorizer will either allow or deny the operation.
+The decisions are of course cached by the authorizer to make sure the performance of the Kafka cluster is not affected.
+
+We didn't developed our own OPA Authorizer for Kafka.
+Instead we are using the existing [OPA Plugin from Bisnode](https://github.com/Bisnode/opa-kafka-plugin).
+
+### Configuring OPA authorizer
+
+Before we configure our Kafka cluster to use OPA, we first need to deploy the OPA server.
+If you don't use OPA yet, you can follow [this guide](https://www.openpolicyagent.org/docs/latest/deployments/#kubernetes) from the OPA documentation.
+
+Once you have OPA running, you can configure the OPA authorizer in your `Kafka` custom resource.
+The basic configuration should looks like this:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    authorization:
+      type: opa
+      url: http://opa.namespace.svc:8181/v1/data/kafka/authz/allow
+    # ...
+```
+
+For using OPA authorizer, you have to set the `type` field to `opa`.
+You also always need to configure the `url`.
+The URL will not just tell it how to connect to the OPA server.
+But it will also specify which policy and rule should be used.
+In the above example, we will connect to the OPA server at `http://opa.namespace.svc:8181`.
+And the `/kafka/authz/allow` part defines that we will use the `allow` decision from the policy from package `kafka.authz`.
+
+There are some additional options which are optional.
+You can configure the decision cache - its capacity and expiration.
+You can also configure super users which will be allowed everything without asking OPA for a decision.
+The full OPA authorizer configuration might look for example like this:
+
+```yaml
+authorization:
+  type: opa
+  url: http://opa.namespace.svc:8181/v1/data/kafka/authz/allow
+  allowOnError: false
+  initialCacheCapacity: 1000
+  maximumCacheSize: 10000
+  expireAfterMs: 60000
+  superUsers:
+    - CN=henri
+    - anwar
+    - CN=wesley
+```
+
+### Examples policies
+
+Explaining the basics of the Rego language and Open Policy Agent policies goes well beyond the scope of this blog post.
+Both is nicely explained in the [OPA documentation](https://www.openpolicyagent.org/docs/latest/).
+But I will show you some examples how you can use OPA with Apache Kafka to demonstrate its strength and versatility.
+The complete policy files from these examples including the instructions how to use them can be found on my [GitHub](https://github.com/scholzj/demo-opa-kafka-authorization).
+
+Additionally, the OPA documentation has a simple demo for [Kafka authorization](https://www.openpolicyagent.org/docs/latest/kafka-authorization/) as well.
+
+### Group based authorization
+
+One of the features users are asking for from time to time is group based authorization.
```

Comment:
```suggestion
One of the features Strimzi users are asking for is group based authorization.
```
I think the frequency is not needed, and to say we are addressing the desire if enough.